### PR TITLE
uncrustify 0.73.0

### PR DESCRIPTION
**Pull request by clason**
Created with `brew bump-formula-pr`.

Includes the patches from https://github.com/uncrustify/uncrustify/pull/3178 and https://github.com/uncrustify/uncrustify/pull/3179 that fix the build failures of the previous attempt https://github.com/Homebrew/homebrew-core/pull/77330 (can be removed for 0.74.0, whenever that comes out).

Some weird build failure on Linux that I cannot debug since I don't have access to a Linux machine, so marking this `macos-only` for now so at least that platform is unblocked. (I don't think the old PR, and the old version, was built on Linux?)

### CODE REVIEW

**Comment by SMillerDev** [comment]
> can be removed for 0.74.0, whenever that comes out

Please add this as a comment above the patch

File: Formula/uncrustify.rb [hunk 1]
```diff
@@ -16,8 +16,13 @@ class Uncrustify < Formula
   end
 
   depends_on "cmake" => :build
+  depends_on :macos
```
Comment (SMillerDev) [inline Formula/uncrustify.rb:19]: Seeing that there is a linux bottle, it worked before the patch.

Comment (clason) [reply Formula/uncrustify.rb:19]: Possibly, but there's no Linux CI on the linked PR?

Comment (Bo98) [reply Formula/uncrustify.rb:19]: Linux CI arrived later, as a part of the merging of homebrew-core and linuxbrew-core.

Comment (clason) [reply Formula/uncrustify.rb:19]: Yes, that much was clear. I can't help but lament the shifting goal posts; makes it significantly harder to contribute to homebrew.

Comment (carlocab) [reply Formula/uncrustify.rb:19]: Let's get rid of this line so we can see what the Linux failure is and try to fix it.

**Comment by carlocab** [comment]
Ok, let's see if a newer GCC helps...

File: Formula/uncrustify.rb (hunk 1 above)
Comment (clason) [reply Formula/uncrustify.rb]: https://github.com/Homebrew/homebrew-core/pull/86327/checks?check_run_id=3781104013#step:7:279 

Looks like some bad compiler setup

**Comment by clason** [comment]
Yep, that did it \o/

**Review by carlocab — APPROVED** [review]
Thanks @clason! We should see about moving that patch to Homebrew/formula-patches. It's a bit large to be embedded. Also not friendly to whitespace-eating editors.

**Comment by clason** [comment]
> Thanks @clason! We should see about moving that patch to Homebrew/formula-patches. It's a bit large to be embedded. Also not friendly to whitespace-eating editors.

I thought about that, but it should be gone by the next release (which, looking now at https://github.com/uncrustify/uncrustify/releases, I expect sometime in November)?

**Comment by clason** [comment]
Thanks @carlocab! As always, I really appreciate your kind and patient help!

**Comment by carlocab** [comment]
No, thank _you_. We couldn't run this project without contributors like you.

> I thought about that, but it should be gone by the next release (which, looking now at https://github.com/uncrustify/uncrustify/releases, I expect sometime in November)?

Probably worth doing anyway -- lots can happen between now and then.

**Comment by clason** [comment]
Fair! I'll see what the format is and open follow-up PRs.

**Comment by clason** [comment]
So how is this actually done? I don't see any PRs on the `formula-patches` repo...

**Pull request closed without merging**